NEW_README.md

```diff
@@ -141,15 +140,155 @@ Run prints its output if either,
+- If task A is triggered by task B, and both A and B are running, a successful completion of B will trigger A to restart.
+- **Triggers are not considered when a task is initially run.** They only affect task restarts.
+- **Long tasks cannot be triggers.** It is invalid to use a long task as a trigger, since long tasks aren't expected to end.
+
+>[!NOTE]
+> Task IDs listed as `dependencies` and `triggers` can cross directory boundaries (e.g., `"css/build"` refers to a task in `./css/tasks.toml`).
+
+##### `watch`
+Defines file paths or globs to monitor for changes. Any detected change triggers a task restart. Examples include:
+- `"."` for the current directory (excluding subdirectories)
+- `"**"` for any change within the working directory
+- Specific paths like `"./src/website/**/*.js"` for targeted file types.
+
+##### `env`
+A map of environment variables provided to the task's execution environment. 
+
+- These variables are available to the command running the task.
+- They are appended to the current environment, overriding any existing variables with the same name.
+- They persist for the duration of the task's execution.
+
+#### `cmd`
+The shell command to execute as the task. It's run in a new bash process:
+
+    $ bash -c "$CMD"
+
+In simple cases, the command can be a single line:
+
+```toml
+[[task]]
+  id = "clean"
+  type = "short"
+  cmd = "go clean -testcache && go clean -modcache"
+```
+
+For more complex commands, or to aid readability, use a multiline string:
+
+```toml
+[[task]]
+  id = "clean"
+  type = "short"
+  cmd = """
+    echo "Cleaning up..."
+    set -x
+
+    rm -rf ./bin
+    go clean -testcache
+    go clean -modcache
+  """
+```
+
 ## Programmatic Use
 
-Run can be used and extended programmatically through its Go API. For more
-information, including a conceptual overview of the architecture, example code,
-and reference documentation, see [the godoc][godoc].
+Run can be used and extended programmatically through its Go API. 
+
+In this small example, we use components from Run to build our own version of the run CLI tool. See cmd/run for the source of the -real- run CLI, which isn't too much more complex. 
+
+```go
+package main
+
+import "github.com/amonks/run/pkg/run"
+
+func main() {
+	tasks, _ := run.Load(".")
+	r, _ := run.RunTask(".", tasks, "dev")
+	ui := run.NewTUI(r)
+
+	ctx := context.Background()
+	uiReady := make(chan struct{})
+
+	go ui.Start(ctx, uiReady, os.Stdin, os.Stdout)
+	<- uiReady
+
+	r.Start(ctx, ui) // blocks until done
+}
+```
+
+For more information, a conceptual overview of the architecture, example code, and reference documentation, see [the godoc][godoc].
 
 [godoc]: https://amonks.github.io/run
 
-# Attribution and License
+## Examples
+
+- [demo](https://github.com/amonks/run/tree/main/examples/demo): A realistic project configuration with multiple tasks. Used to demonstrate Run's capabilities.
```

Comment:
TODO: consolidate examples

- the existing demo example is dumb and weird, and should be deleted
- the golang-app example is conceptually heavy, and should not be merged
- I haven't looked at the hello example; maybe it's great!